bdikit/functional_api.py

```diff
@@ -0,0 +1,138 @@
+from enum import Enum
+from os.path import join, dirname
+from typing import Union, Type, List, Optional
+import pandas as pd
+from bdikit.download import get_cached_model_or_download
+from bdikit.mapping_algorithms.column_mapping.algorithms import (
+    BaseColumnMappingAlgorithm,
+    SimFloodAlgorithm,
+    ComaAlgorithm,
+    CupidAlgorithm,
+    DistributionBasedAlgorithm,
+    JaccardDistanceAlgorithm,
+    GPTAlgorithm,
+)
+from bdikit.mapping_algorithms.value_mapping.value_mappers import ValueMapper
+from bdikit.mapping_algorithms.scope_reducing._algorithms.contrastive_learning.cl_api import (
+    ContrastiveLearningAPI,
+)
+
+GDC_DATA_PATH = join(dirname(__file__), "./resource/gdc_table.csv")
+
+
+class ColumnMappingMethod(Enum):
```

Comment:
Would this class not be better off at mapping_algorithms/column_mapping/algorithms.py? Maybe a factory pattern could take care of such cases; as they will appear more often.

Reply:
Yes, we should probably move it either to algorithms.py or to the `column_mapping` module level, but I added it here for now to avoid confusion with the other enum that exists in the `column_mapping_manager.py` class. The static `ColumnMappingMethod.get_instance()` method already works like a factory method, we can evolve it when necessity arises.